determined-ai/determined
# build: setup `Pre-Commit`

## Description

- [DET-8095]
- Setup [pre-commit](https://pre-commit.com/) for entire repo and setup `lint-staged` for web


## How to Setup

- Run `make get-deps-pip`
- Run `make set-config-dev`
- Change file in `webui/react/src` folder
- Commit to run pre-commit

## Test Plan
- Check if all configurations are correct (especially pre-commit)
- Check if pre-commit is triggered correctly when specific files in web folder are staged and about to be committed



## Commentary (optional)




## Checklist
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.





[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[DET-8093]: https://determinedai.atlassian.net/browse/DET-8093?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DET-8095]: https://determinedai.atlassian.net/browse/DET-8095?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
